invenio_rdm_records/services/components/pids.py

```diff
@@ -246,10 +243,26 @@ def edit(self, identity, draft=None, record=None):
         # FIXME: this should be moved to marshmallow level
         # validation happens before to be able to fail before reserving pids
         # it means a double looping but validation should eventually be moved
-        # to marshmellow see invenio-rdm-records/issues/796
-        errors = {}
+        # to marshmallow see invenio-rdm-records/issues/796
+        errors = []
         self._validate_pids_schemes(record_pids)
         self._validate_pids(record_pids, record, errors)
         if errors:
             raise ValidationError(message=errors)
         draft.pids = record_pids
+
+    def post_publish(self, identity, draft=None, record=None):
+        """Post publish handler."""
+        record_pids = record.get('pids', {})
+        for scheme, pid_dict in record_pids.items():
+            provider_name = pid_dict.get("provider")
+            identifier_value = pid_dict.get("identifier")
+            provider = current_rdm_records.records_service.pids.get_provider(
```

Comment:
minor: just for code readability maybe we can assignthe service before the for loop.

```python
service = current_rdm_records.records_service
for ...
    service.pids....
```